lua/telescope/config.lua

```diff
@@ -113,6 +113,10 @@ function config.set_defaults(defaults)
   set("border", {})
   set("borderchars", { '─', '│', '─', '│', '╭', '╮', '╯', '╰'})
 
+  set("hide_filename", false)
```

Comment:
can we change this to `hide_path` if we have the other values?

my other concern is that we're going to explode into many different `X_path` options... I wonder if there is a way we can get ahead of this but still keep it simple for people to use. perhaps some ENUM style item for path displaying? I'm not sure.

Reply:
I was going with `hide_filename` because some of the pickers already use it, but `hide_path` sounds better.

Enum in this case would work, since the options are mutually exclusive, but its possible that we include additional paths that may break that assumption. For example, I thought that we could maybe add a `absolute_path` option, in case user prefers absolute/relative paths. If this were the case, we would need separate enum values for `shorten_path=true`, `absolute_path=true`, and `shorten_path && absolute_path`. 

I ended up not including example option, since I'm not sure if people want that. But maybe there are other cases that breaks mutual exclusivity, I just can't think of them. If there's nothing else, I can go with the enum idea.

Reply:
We could do "enum" style, that is more like "flags".

So we can have them be binary flags see if the flag is set. that would allow us the flexibility in the future and for the 99% use case, users just say "telescope.enum_things.path_options.HIDE_PATH" or something.

I can think about this more. gotta run right now

Reply:
I created another commit that tries out adhoc enums. Configs can be set with something like

```
local defaults = {}
defaults[require('telescope.types').path_display_options.SHORTEN_PATH] = true
require('telescope').setup{
    defaults = defaults
}
```

I'm honestly not feeling like this makes the world better, but maybe it can inspire some ideas.

Reply:
Another idea. Could have user specify something like:

```
local defaults = {}
defaults.path_display_options = {
  require(tele.configs).SHORTEN_PATH,
  require(tele.configs).SOME_OTHER_OPT,
}
require('telescope').setup{ defaults=defaults }
```
And the path transformation function can just look through the given options. Again, not sure how nice this is to use. Open to any opinions.